### PR TITLE
OMI-435: write changelog for release/0.3.0

### PR DESCRIPTION
**Pull request by cereniyim**
--- don't change below this line ---
OMI-435 #time 31m

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -2,7 +2,26 @@
...
+  - `ion_mode`: `positive` or `negative`
+- caching of successful spectrum predictions:
+  - if predictions for spectra fails, they can be called by `faied_spectra()` 
+  - cache contains successful predictions of spectra, implemented `reset_cache()` 
+to reset it
+- moved plotting functionality to `omigami.plotting`
+  - `MoleculePlotter` is available in `omigami.plotting` instead of `omigami`
+- added `SpectraComparisonPlotter().mirror_plot` available in `omigami.plotting`:
+  - It takes two spectra returned from `match_spectra` and creates a mirror plot 
+with spectrum 1 on top (blue), and spectrum 2 on the bottom (red)
```
Comment (carloszanella) [inline Changelog.md]: I don't like this structure. I think something like this is better:

# Releases

## v0.3.0
`Release date: 2022-01-10`

### New Features
- Feature name: feature description

### Bugfixes:
- Bug/bugfix description

### Breaking Changes
- what doesnt work anymore

Comment (cereniyim) [reply Changelog.md]: ok I will change

**Review by bernardolk — APPROVED** [review]

File: Changelog.md [hunk 2]
```diff
@@ -1,8 +1,52 @@
...
-### Summary of changes
-- to be summarized
-  
-### Details
-- to be detailed
\ No newline at end of file
+## Releases
+***
+v0.3.0
+***
```
Comment (carloszanella) [inline Changelog.md]: I think these 4 lines should be ## v0.3.0. The rendering looks weird

File: Changelog.md [hunk 3]
```diff
@@ -1,8 +1,52 @@
...
+Previous error messages lacking the cause of the error. E.g. 
+> InternalServerError: Something went wrong, the requested service is 
+> probably unavailable at the moment. Please try again later or contact DataRevenue 
+> for more information.
+
+Current error message displays the cause. E.g.
+> InternalServerError: Error in generating the prediction: SpectraMatchingError
+> 500: 'No data found from filtering with precursor MZ for spectra at indices 
+> [5, 6, 7, 39, 40, 42]. Try increasing mz_range filtering.'. Please try again
+> later or contact DataRevenue for more information.
```
Comment (carloszanella) [inline Changelog.md]: No need to get into this detail. I think this part is not necessary. Maybe just how it works now as an example, but wrap it in code formatting

```
like this
```

File: Changelog.md [hunk 4]
```diff
@@ -1,8 +1,52 @@
...
+module. It takes two spectra returned from `match_spectra()` and  creates a mirror 
+plot with spectrum 1 on top (blue), and spectrum 2 on the bottom (red).
+- **Caching functionality**: Both `Spec2Vec` and `MS2DeepScore`objects cache 
+successful predictions, also there are 2 new methods:
+  - if predictions for spectra fails, they can be called by `faied_spectra()`.
+  - cache contains successful predictions of spectra, implemented `reset_cache()`
+    to reset it.
+- **Implement SpectrumDataFrameHelper:** Accessible from `omigami.utilities`, 
+this class serves as a helper to create and manipulate Pandas DataFrame versions 
+of Spectrum objects, useful for some plotting functions (such as `mirror_plot()`).
```
Comment (carloszanella) [inline Changelog.md]: please add about returning all metadata in the response now

File: Changelog.md [hunk 5]
```diff
@@ -1,8 +1,52 @@
...
+
+### Bugfixes
+- **More detailed error messages**: Omigami was unable to catch errors from 
+backend during `match_spectra()` call. Omigami properly handles errors and 
+displays them.
```
Comment (carloszanella) [inline Changelog.md]: Please add about the bugs regarding precursor_mz/pepmass and ExactMass/parent_mass. They should both be in Jira

File: Changelog.md [hunk 6]
```diff
@@ -1,8 +1,52 @@
...
+***
+Release Date: 2022-01-10
+
+### New Features
+- **Mirror plot**: Add `SpectraComparisonPlotter().mirror_plot()` to `omigami.plotting` 
+module. It takes two spectra returned from `match_spectra()` and  creates a mirror 
+plot with spectrum 1 on top (blue), and spectrum 2 on the bottom (red).
+- **Caching functionality**: Both `Spec2Vec` and `MS2DeepScore`objects cache 
+successful predictions, also there are 2 new methods:
+  - if predictions for spectra fails, they can be called by `faied_spectra()`.
```
Comment (carloszanella) [inline Changelog.md]: it is not a method, it is a property. just `failed_spectra`

File: Changelog.md (hunk 2 above)
Comment (cereniyim) [reply Changelog.md]: you mean we need 4 starts to draw a line properly?

Comment (cereniyim) [reply Changelog.md]: ok I updated headings

**Review by carloszanella — APPROVED** [review]

**Review by bernardolk — APPROVED** [review]

**Pull request merged**